### ISSUE TITLE
validation of protected properties is broken when class is merged with interface from another file

### ISSUE THREAD

**Issue by vladima** [Microsoft/TypeScript#6762]
falling test cases
```ts
// @filename: a.ts
class A {
    protected foo() {}
}
// @filename: b.ts
interface A { }

class B extends A {
    protected foo() {}
}
```

```ts
// @module: amd

// @filename: a.ts
export class A {
    protected protected: any;

    protected setProtected(val: any) {
        this.protected = val;
    }
}

// @filename: b.ts
import {A} from './a';

declare module "./a" {
    interface A { }
}

export class B extends A {
    protected setProtected() {

    }
}
```

Both tests should pass without errors however now the same error is reported in both cases:
`Class 'B' incorrectly extends base class 'A'.`

**Comment by masaeedu** [comment]
@vladima Will this change also be ported to the 1.8 release?

**Comment by mhegazy** [comment]
yes. please see https://github.com/Microsoft/TypeScript/pull/6786

**Comment by masaeedu** [comment]
Awesome, thanks!
